### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by dnalborczyk**
| Q                        | A 
| ------------------------ | --- |
| Fixed Issues?            | |
| Patch: Bug Fix?          | |
| Major: Breaking Change?  | |
| Minor: New Feature?      | |
| Tests ~Added~ + Pass?      |:white_check_mark: `make build`<br>:white_check_mark: `make test`<br>:white_check_mark: `make flow`<br> :white_check_mark: `make lint` |
| Documentation PR Link    | 
| Any Dependency Changes?  | `babel` and related packages: v7.0 beta 54<br>`babel-loader` v8.0 beta 4<br>`flow-bin` v0.77 _(minor)_<br>`eslint` v5.2 _(minor)_<br>`browserify` v16.2 _(major)_ |
| License                  | MIT |

update dev dependencies to latest versions, divided in 3 PRs

`browserify` update has a dependency on a module with an intentionally broken package.json which throws off `flow`. (added to ignore section of flow config).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,29 +10,29 @@
...
   },
   "devDependencies": {
-    "@babel/cli": "7.0.0-beta.52",
-    "@babel/core": "7.0.0-beta.52",
-    "@babel/plugin-transform-modules-commonjs": "7.0.0-beta.52",
-    "@babel/preset-env": "7.0.0-beta.52",
-    "@babel/preset-flow": "7.0.0-beta.52",
-    "@babel/preset-stage-0": "7.0.0-beta.52",
-    "@babel/register": "7.0.0-beta.52",
+    "@babel/cli": "^7.0.0-beta.54",
```
Comment (nicolo-ribaudo) [inline package.json]: Please remove `^`

Comment (dnalborczyk) [reply package.json]: on it

File: packages/babel-parser/src/plugins/typescript.js [hunk 2]
```diff
@@ -1539,11 +1539,13 @@ export default (superClass: Class<Parser>): Class<Parser> =>
...
       // https://github.com/Microsoft/TypeScript/pull/16040
       if (this.state.value === "interface") {
-        return this.tsParseDeclaration(
+        const result = this.tsParseDeclaration(
           this.startNode(),
           this.state.value,
           true,
         );
+
+        if (result) return result;
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/plugins/typescript.js]: This changes the behavior of Babylon. Is it intended? If so, can you extract it to a separate PR and add a test?

Comment (existentialism) [reply packages/babel-parser/src/plugins/typescript.js]: Yeah, new errors detected from latest Flow, basically `tsParseDeclaration` can return `undefined` and the typing of `parseExportDefaultExpression` doesn't allow it.

Comment (nicolo-ribaudo) [reply packages/babel-parser/src/plugins/typescript.js]: Nah it's ok
